script/hassfest/requirements.py

```diff
@@ -0,0 +1,154 @@
+"""Validate requirements."""
+import operator
+import subprocess
+import sys
+from typing import Dict, Set
+
+from script.gen_requirements_all import COMMENT_REQUIREMENTS
+from stdlib_list import stdlib_list
+from tqdm import tqdm
+
+from homeassistant.const import REQUIRED_PYTHON_VER
+import homeassistant.util.package as pkg_util
+
+from .model import Config, Integration
+
+SUPPORTED_PYTHON_TUPLES = [
+    REQUIRED_PYTHON_VER[:2],
+    tuple(map(operator.add, REQUIRED_PYTHON_VER, (0, 1, 0)))[:2],
+]
+SUPPORTED_PYTHON_VERSIONS = [
+    ".".join(map(str, version_tuple)) for version_tuple in SUPPORTED_PYTHON_TUPLES
+]
+STD_LIBS = {version: set(stdlib_list(version)) for version in SUPPORTED_PYTHON_VERSIONS}
+
+
+def normalize_package_name(requirement: str) -> str:
+    """Return a normalized package name from a requirement string."""
+    requirement = requirement.split(" ")[-1]  # remove potential pip argument
```

Comment:
Maybe use a regular expression to get it out?

`^(?:--.+\s)?([\w\d-_]+).*==.+$` (group 1 will have the package)

https://regexr.com/5b1em